test/archethic/mining/distributed_workflow_test.exs

```diff
@@ -968,7 +998,279 @@ defmodule Archethic.Mining.DistributedWorkflowTest do
           end
 
           assert_receive :replication_done
+          refute_receive :validation_error
       end
     end
+
+    test "should not replicate if there is a validation error", %{tx: tx} do
+      validation_context = create_context(tx)
+
+      validation_stamp = create_validation_stamp(validation_context)
+      validation_stamp = %ValidationStamp{validation_stamp | error: :invalid_pending_transaction}
+
+      context =
+        validation_context
+        |> ValidationContext.add_validation_stamp(validation_stamp)
+
+      me = self()
+
+      MockClient
+      |> stub(:send_message, fn
+        _, %Ping{}, _ ->
+          {:ok, %Ok{}}
+
+        _, %GetUnspentOutputs{}, _ ->
+          {:ok, %UnspentOutputList{unspent_outputs: []}}
+
+        _, %ValidationError{}, _ ->
+          send(me, :validation_error)
+          {:ok, %Ok{}}
+
+        _, %GetTransactionSummary{}, _ ->
+          {:ok, %NotFound{}}
+
+        _, %GetTransaction{}, _ ->
+          {:ok, %Transaction{}}
+      end)
+
+      {:ok, coordinator_pid} =
+        Workflow.start_link(
+          transaction: context.transaction,
+          welcome_node: context.welcome_node,
+          validation_nodes: [context.coordinator_node | context.cross_validation_nodes],
+          node_public_key: context.coordinator_node.last_public_key
+        )
+
+      :sys.replace_state(coordinator_pid, fn {:coordinator, %{context: _}} ->
+        {:wait_cross_validation_stamps, %{context: context}}
+      end)
+
+      Workflow.add_cross_validation_stamp(
+        coordinator_pid,
+        %CrossValidationStamp{
+          signature:
+            Crypto.sign(
+              [ValidationStamp.serialize(context.validation_stamp), <<1>>],
+              elem(@publickey1, 1)
+            ),
+          node_public_key: elem(@publickey1, 0),
+          inconsistencies: [:signature]
```

Comment:
I think in this text it should not have the cross validation stamp errors but only the pending one.

Reply:
good catch